src/Lean/Meta/Tactic/Apply.lean

```diff
@@ -50,8 +51,18 @@ def getExpectedNumArgs (e : Expr) : MetaM Nat := do
   let (numArgs, _) ← getExpectedNumArgsAux e
   pure numArgs
 
-private def throwApplyError {α} (mvarId : MVarId) (eType : Expr) (targetType : Expr) : MetaM α :=
-  throwTacticEx `apply mvarId m!"failed to unify{indentExpr eType}\nwith{indentExpr targetType}"
+private def throwApplyError {α} (mvarId : MVarId) (eType : Expr) (targetType : Expr) : MetaM α := do
+  let explanation := MessageData.lazy
+    (f := fun
+      | none =>
+        return f!"{eType}\nwith{targetType}"
```

Comment:
```suggestion
        return f!"{indentExpr eType}\nwith{indentExpr targetType}"
```
Perhaps?

Reply:
That only exists for MessageData not Format. But the whole case is a bit bogus and hopefully dead code, we should always have a context here, I hope. Maybe I'll make this an error and see if anyone hits this code path.